### PR TITLE
OCT-1508 OCT-1510 Require address in allocate in a param: server & client

### PR DESCRIPTION
**Pull request by paulperegud**
Previously we relied on erecover. This doesn't work for EIP-1271.

### CODE REVIEW

**Review by leoni-q — APPROVED** [review]

File: backend/app/infrastructure/routes/allocations.py [hunk 1]
```diff
@@ -62,6 +62,10 @@
     "UserAllocationRequest",
     {
         "payload": fields.Nested(allocation_payload),
+        "user_address": fields.String(
```
Comment (aziolek) [inline backend/app/infrastructure/routes/allocations.py:65]: @paulperegud: please change to camelCase, unifying all the fields across the endpoint and most of the stack.
```suggestion
        "userAddress": fields.String(
```

**Review by aziolek — CHANGES_REQUESTED** [review]

File: backend/app/infrastructure/routes/allocations.py (hunk 1 above)
Comment (aziolek) [reply backend/app/infrastructure/routes/allocations.py:65]: [The client is now adjusted and sends camelCased `userAddress`](https://github.com/golemfoundation/octant/pull/112/commits/cb639db6379f6b01d36126f3a688844528bff464).

**Review by kgarbacinski — CHANGES_REQUESTED** [review]
please align ```handle_allocate``` from ```app.infrastracture.events``` with new changes that call controller.allocate in the old way

File: backend/tests/conftest.py [hunk 2]
```diff
@@ -311,9 +311,14 @@ def get_allocation_nonce(self, address: str) -> int:
...
         return json.loads(rv)["allocationNonce"]
 
-    def allocate(self, payload: dict, signature: str) -> int:
+    def allocate(self, payload: dict, user_address: str, signature: str) -> int:
         rv = self._flask_client.post(
-            "/allocations/allocate", json={"payload": payload, "signature": signature}
+            "/allocations/allocate",
+            json={
+                "payload": payload,
+                "user_address": user_address,
```
Comment (leoni-q) [inline backend/tests/conftest.py:319]:
```suggestion
                "userAddress": user_address,
```

**Review by leoni-q — APPROVED** [review]

**Review by leoni-q — APPROVED** [review]

**Review by kgarbacinski — APPROVED** [review]

**Pull request merged**